src/crimson/osd/pg.cc

```diff
@@ -513,6 +513,53 @@ Context *PG::on_clean()
   return nullptr;
 }
 
+seastar::future<> PG::clear_ondisk_temp_objs()
+{
+  logger().info("{} {}", __func__, pgid);
+  return seastar::do_with(
+    ghobject_t(),
+    ceph::os::Transaction(),
+    (uint64_t)0,
+    [this](auto &_next, auto &t, auto &removed) {
+    return seastar::repeat([&_next, this, &t, &removed] {
+      return shard_services.get_store().list_objects(
+        coll_ref,
+        _next,
+        ghobject_t::get_max(),
+        local_conf()->osd_target_transaction_size
+      ).then([&_next, this, &t, &removed](auto p) {
+        auto &objs = std::get<0>(p);
+        auto &next = std::get<1>(p);
```

Comment:
This can be a good opportunity to avoid callbacks with co routines:
```
  auto [objs, next] = co_await shard_services.get_store()->list_objects(...);
```